presto-main/src/main/java/com/facebook/presto/cost/ComparisonStatsCalculator.java

```diff
@@ -217,7 +218,25 @@ private static PlanNodeStatsEstimate expressionToExpressionNonEquality(
             Optional<Symbol> right,
             SymbolStatsEstimate rightStats)
     {
-        return PlanNodeStatsEstimateMath.differenceInStats(inputStatistics, expressionToExpressionEquality(inputStatistics, left, leftStats, right, rightStats));
+        double nullsFilterFactor = (1 - leftStats.getNullsFraction()) * (1 - rightStats.getNullsFraction());
+        PlanNodeStatsEstimate inputNullsFiltered = inputStatistics.mapOutputRowCount(size -> size * nullsFilterFactor);
+        SymbolStatsEstimate leftNullsFiltered = leftStats.mapNullsFraction(nullsFraction -> 0.0);
```

Comment:
capping is done as separate rule, on the output of filter estimation.
We probably want capping while we compute stats for complex filter expression. But currently it is not happening anywhere.